src/Logger/QueueLogger.php

```diff
@@ -4,30 +4,83 @@
 
 use Monolog\Logger;
 use Monolog\Handler\RedisHandler;
+use Monolog\Handler\StreamHandler;
 use Monolog\Formatter\LogstashFormatter;
+use MessageQueuePHP\Message\Message;
+use MessageQueuePHP\Adapter\AdapterInterface;
+use MessageQueuePHP\Config\ConfigInterface;
 use Predis\Client as RedisClient;
 
-class QueueLogger
+class QueueLogger implements AdapterInterface
 {
-	private $logger;
+    /**
+     * @var [Monolog\Logger]
+     */
+    private $logger;
 
-	public function __construct($channel, $key, $host, $port)
-	{
-		$redisClient = new RedisClient([
+    /**
+     * @param string $channel
+     * @param string $key
+     * @param string $host
+     * @param string $port
+     */
+    public function __construct(ConfigInterface $config)
+    {
+        $configLogger = $config->getConfig();
+
+        $host    = $configLogger['logger']['host'];
+        $port    = $configLogger['logger']['port'];
+        $key     = $configLogger['logger']['key'];
+        $channel = $configLogger['logger']['channel'];
+        $path    = $configLogger['logger']['path'];
+
+        $redisClient  = new RedisClient([
                     'scheme' => 'tcp',
                     'host'   => $host,
                     'port'   => $port
                 ]);
 
-		$redisHandler = new RedisHandler($redisClient, $key);
-		$formatter    = new LogstashFormatter($channel);
-		$redisHandler->setFormatter($formatter);
+        try {
+            $redisClient->ping();
+            $handler   = new RedisHandler($redisClient, $key);
+            $formatter = new LogstashFormatter($channel);
+            $handler->setFormatter($formatter);
+        }catch (Exception $e) {
+            $handler = new StreamHandler($path);
+        }
+
+        $this->logger = new Logger($channel, array($handler));
+    }
+
+    /**
+     * @param string $message
+     * @param string $level
+     */
+    public function setMessage($message, $level)
+    {
+        $this->logger->$level($message);
+    }
 
-		$this->logger = new Logger($channel, array($redisHandler));
-	}
+    /**
+     * @param  Message $message
+     * @param  string $queue
+     * @param  string $exchange
+     * @return void
+     */
+    public function send(Message $message, $queue, $exchange)
+    {
+        $this->logger->info($message->getPayload(), [$queue]);
```

Comment:
$this->setMessage('Message Queue is not Working', 'warning', ['message' => $message->getPayload(), 'queue' => $queue, 'exchange' => $exchange])